applications/ParticleMechanicsApplication/custom_elements/updated_lagrangian_UP.cpp

```diff
@@ -1037,7 +1017,7 @@ void UpdatedLagrangianUP::CalculateMassMatrix( MatrixType& rMassMatrix, const Pr
         for (IndexType i = 0; i < number_of_nodes; ++i) {
             for (IndexType k = 0; k < dimension; ++k)
             {
-                const IndexType index = i * dimension + k;
+                const IndexType index = i * (dimension + 1) + k;
                 rMassMatrix(index, index) = N[i] * mMP.mass;
             }
         }
```

Comment:
It's really hard to follow all the changes in the UP-Element. We need to be sure that it still works - did you run further validaton examples?

Reply:
I see. Here it was just a problem deriving from the fact that a degree of freedom is added in the UP element but the Mass Matrix dimension is not updated accordingly. For the complete formulation, I attach my thesis hoping it will describe better any change adopted
[tesi_alessandro_contri.pdf](https://github.com/KratosMultiphysics/Kratos/files/7942023/tesi_alessandro_contri.pdf)

Reply:
Nice document